infrastructure/api/sse-api.yml

```
@@ -317,10 +318,47 @@ Resources:
     Properties:
       Handler: backend/api/src/handlers/auth/update-client.updateClientInRegistryHandler
       Description: Updates a client using an HTTP endpoint
+      KmsKeyArn: !Ref EnvironmentEncryptionKey
       Environment:
         Variables:
           AUTH_REGISTRATION_BASE_URL: !Ref AuthRegistrationBaseUrl
           AUTH_API_KEY: "{{resolve:ssm:/self-service/api/auth-api-key}}"
+      Policies:
+        - Statement:
+            - Effect: Allow
+              Action:
+                - "kms:Decrypt"
+                - "kms:Encrypt"
+              Resource:
+                - !Ref EnvironmentEncryptionKey
+  
+ # KMS configuration for auth kms key
+  EnvironmentEncryptionKey:
+    Type: AWS::KMS::Key
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    # checkov:skip=CKV_AWS_7: "Ensure rotation for customer created CMKs is enabled"
+    Properties:
+      Description: Key for auth key encrypt/decrypt
+      Enabled: TRUE
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: kms:*
+            Resource: '*'
+      KeySpec: ECC_NIST_P256
+      KeyUsage: ENCRYPT_DECRYPT
+
+  AuthKmsKeyAlias:
+    Type: AWS::KMS::Alias
+    Properties:
+      AliasName: !Sub "sse_auth_kms_Key"
+      TargetKeyId: !Ref EnvironmentEncryptionKey
```

```suggestion
  EnvironmentEncryptionKeyAlias:
    Type: AWS::KMS::Alias
    Properties:
      AliasName: api-lambda-environment-encryption-key
      TargetKeyId: !Ref EnvironmentEncryptionKey
```
We'll need to use this key for other lambdas to encrypt their environments in the future, so let's separate it from Auth.

---

Alias name must start with `alias` i.e
`alias/api-lambda-environment-encryption-key`